src/ert/_c_wrappers/enkf/_config_content_as_dict.py

```diff
@@ -0,0 +1,112 @@
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from ert._c_wrappers.enkf.config_keys import ConfigKeys
+
+if TYPE_CHECKING:
+    from ert._c_wrappers.config import ConfigContent
+
+# keywords that have one argument and
+# the last occurrence of the keyword in the file
+# is the value of which we use.
+# ie. if the config contains:
+#
+# MAX_SUBMIT 2
+# MAX_SUBMIT 3
+# ert will use the value 3 for MAX_SUBMIT
+SINGLE_VALUE_SINGLE_ARG_KEYS = [
+    ConfigKeys.ALPHA_KEY,
+    ConfigKeys.ANALYSIS_SELECT,
+    ConfigKeys.CONFIG_DIRECTORY,
+    ConfigKeys.DATAROOT,
+    ConfigKeys.DATA_FILE,
+    ConfigKeys.ECLBASE,
+    ConfigKeys.ENSPATH,
+    ConfigKeys.GEN_KW_EXPORT_NAME,
+    ConfigKeys.GEN_KW_TAG_FORMAT,
+    ConfigKeys.GRID,
+    ConfigKeys.HISTORY_SOURCE,
+    ConfigKeys.ITER_CASE,
+    ConfigKeys.ITER_COUNT,
+    ConfigKeys.ITER_RETRY_COUNT,
+    ConfigKeys.JOBNAME,
+    ConfigKeys.JOB_SCRIPT,
+    ConfigKeys.LICENSE_PATH,
+    ConfigKeys.MAX_RESAMPLE,
+    ConfigKeys.MAX_RUNTIME,
+    ConfigKeys.MAX_SUBMIT,
+    ConfigKeys.MIN_REALIZATIONS,
+    ConfigKeys.NUM_CPU,
+    ConfigKeys.NUM_REALIZATIONS,
+    ConfigKeys.OBS_CONFIG,
+    ConfigKeys.QUEUE_SYSTEM,
+    ConfigKeys.RANDOM_SEED,
+    ConfigKeys.REFCASE,
+    ConfigKeys.RERUN_KEY,
+    ConfigKeys.RERUN_START_KEY,
+    ConfigKeys.RUNPATH,
+    ConfigKeys.RUNPATH_FILE,
+    ConfigKeys.STD_CUTOFF_KEY,
+    ConfigKeys.STOP_LONG_RUNNING,
+    ConfigKeys.TIME_MAP,
+    ConfigKeys.UPDATE_LOG_PATH,
+]
+
+# Like SINGLE_VALUE_SINGLE_ARG_KEYS but
+# the keyword can take more than one argument, ie.
+# SCHEDULE_PREDICTION_FILE filename  <parameters:> <init_files:>
+SINGLE_VALUE_MULTI_ARG_KEYS = [
+    ConfigKeys.SCHEDULE_PREDICTION_FILE,
+]
+
+# Keywords that can occur more than once, but
+# the number of arguments is always one.
+MULTI_VALUE_SINGLE_ARG_KEYS = [
+    ConfigKeys.WORKFLOW_JOB_DIRECTORY,
+    ConfigKeys.INSTALL_JOB_DIRECTORY,
+]
+
+# Some keys have been split prematurely and must been
+# joined back, ie.
+
+# QUEUE_OPTION LOCAL opt the value of the option
+
+# Should be interpreted as having three arguments:
+# ["LOCAL", "opt", "the value of the option"]
+JOIN_KEYS = [
+    (ConfigKeys.QUEUE_OPTION, 2),
+    (ConfigKeys.FORWARD_MODEL, 0),
+]
+
+
+def config_content_as_dict(
```

Comment:
I think this function is trying to do a bit much. Would it be too bad to have code like:
```python
content_dict  = config_content_as_dict(site_config_content)
if user_config_content is not None:
   content_dict.update(config_content_as_dict(user_config_content))
```
but it might be that `   content_dict.update(config_content_as_dict(user_config_content))` does not really work due to having very deep complicated dictionaries.

Reply:
Yes, it wouldn't really work because we don't do a straight `update`. If there are lists in there we sometimes want to append to those lists.

Reply:
I will add some tests that should make it clear.

Reply:
Tests added now